feat(autoware*.repos): remove *_launch refs and bump autoware_launch to 0.43.1

xmfcx opened the pull request:
## Description

Related:
- https://github.com/autowarefoundation/autoware/issues/5912#issuecomment-2743482983

Merge with:
- https://github.com/autowarefoundation/autoware_launch/pull/1369

## How was this PR tested?

## Notes for reviewers

None.

## Effects on system behavior

None.

xmfcx committed:
feat(autoware*.repos): remove *_launch refs

Signed-off-by: M. Fatih Cırıt <mfc@autoware.org>

mitsudome-r commented:
I think we should update the target branch for autoware_launch to 0.43.1 in this PR.

xmfcx committed:
bump autoware_launch version

Signed-off-by: M. Fatih Cırıt <mfc@autoware.org>

youtalk reviewed (approved):
Great works! LGTM

xmfcx committed:
feat(Dockerfile): update sensor_kit and vehicle refs

Signed-off-by: M. Fatih Cırıt <mfc@autoware.org>

xmfcx commented:
Thanks @youtalk -san!

I've also updated the following:
https://github.com/autowarefoundation/autoware/pull/5967/commits/9ad32532201dcefce6e3ab0e8f11df96296869eb

The refs to `sensor_kit` and `vehicle` folders under `src/` are now included within the `src/launcher/autoware_launch/` directory.

I've updated the Dockerfile accordingly. It already copies the launcher folder so I've removed the other copy directives.
